query-engine/connectors/sql-query-connector/src/database/sqlite.rs

```diff
@@ -38,30 +38,13 @@ impl FromSource for Sqlite {
             .map_err(|sql_error| sql_error.into_connector_error(&connection_info))?;
 
         let file_path = params.file_path;
+        let url = database_str.split('?').next();
 
-        let url_with_db = {
-            let db_name = std::path::Path::new(&file_path)
-                .file_stem()
-                .and_then(|stem| stem.to_str())
-                .ok_or_else(|| invalid_file_path_error(&file_path, &connection_info))?
-                .to_owned();
+        if url.is_none() || std::path::Path::new(url.unwrap()).file_stem().is_none() {
+            return Err(invalid_file_path_error(&file_path, &connection_info));
+        }
```

Comment:
Isn't there a more rust idiomatic way to express that?

Reply:
You could use the Url crate (should be loaded in the connectors already) instead and use one of their methods to get the path (or whatever exactly you need here).

Reply:
The crate doesn't work here because it kind of messes up the file paths. We used to use that, but had too many issues and go with `PathBuf` nowadays...

Reply:
This would be a bit shorter:
```rust
if url.map(Path::new).and_then(|url| url.file_stem()).is_nume() {
    return Err(invalid_file_path_error(&file_path, &connection_info));
}
```

Reply:
I was initially mostly referring to the `if X || Y { return Err() }` - Maybe the question is more: Is there a more functional way of returning the error.